episodes/42-software-reuse.md

```diff
@@ -252,9 +271,145 @@ which may be held within other Markdown files within the repository or elsewhere
 
-### Other Documentation
+### Generate and Deploy Documentation using MKDocs
+
+[MKDocs](https://www.mkdocs.org/) generate project documentation as a static website from Markdown files. The website can then be hosted on GitHub Pages or other static site hosting services, providing a user-friendly interface for accessing the documentation.
+
+We can install MKDocs package using `pip`. Here we also install a plugin `mkdocstrings`, which will be used later.
+We advise you to do this within a virtual environment you created before:
+
+```bash
+pip install mkdocs mkdocstrings[python]
+```
+
+By default, `mkdocstrings` does not provide support for a specific language. Therefore, we specify `[python]` to install extra dependencies of `mkdocstrings` for Python language support.
+
+After installation, you can intialize a new MKDocs project in our Python project:
+
+```bash
+mkdocs new .
+```
+
+This will create the two files in your project: `mkdocs.yml` and `docs/index.md`. The first file `mkdocs.yml` is the configuration file for your documentation site. It serves as the central configuration hub for your MKDocs documentation. It tells MKDocs how to structure your documentation site, which plugins and themes to use,
+how to organize navigation, etc.
+
+`docs/index.md` is the main page of your documentation. It is usually the landing page of your documentation site.
+
+Let's first look at the `mkdocs.yml` file. It is almost empty now. We can edit it with the following basic configurations:
+
+```yaml
+site_name: Inflam
+
+nav:
+  - Overview: index.md
+
+plugins:
+  - search
+  - mkdocstrings
+```
+
+Here we give a name to our documentation site, `Inflam`. We set up the navigation menu with one item `Overview` that links to `index.md`. We also enable two plugins, `search` to provide search functionality in the documentation site, and `mkdocstrings` to automatically generate API reference documentation from Python docstrings, which we will see later.
+
+We can try to render the documentation site locally and see how it looks like:
+
+```bash
+mkdocs serve
+```
+
+This will start to build a local static documentation site and serve it at a local web server. 
+By default, it will be available at `http://127.0.0.1:8000/`, which will also show in the terminal output.
+You can open this URL in your web browser to view the documentation site.
+
+The documentation site now consists some default content about MKDocs. It is rendered from the `docs/index.md` file. Let's edit this file to add some relevant content about our project. For simplicity, we can borrow the content from our `README.md` file.
+
+::::::::::::::::::::::::::::::::: challenge
+
+### Exercise: Update Documentation Content
+
+Modify `docs/index.md` with the same content as your `README.md` file.
+Render the documentation site locally again with `mkdocs serve`.
+Check how it looks like in your web browser.
+
```

Comment:
Maybe provide the solution as well (linking the README in the Mkdocs config)? Also, I have noticed that a "challenge" gets rendered as a "discussion" if the solution is not provided (not a big deal).